README.md

```diff
@@ -38,11 +62,94 @@ This has to be added using
 Pkg.clone("https://github.com/marcusps/ExpmV.jl.git")
 ```
 
-**Usage**
-
-QuDynamics.jl
-
+Usage
+-----
 The current package can be used by
 ```julia
 Pkg.clone("https://github.com/JuliaQuantum/QuDynamics.jl.git")
 ```
+
+We could start using the package by :
+```julia
+using QuDynamics
+```
+
+`QuPropagator` is the starting point in order to compute or plot, related entities like states at different time steps, expectation values of an observable and so on. `QuProagator` is the  one which takes
+in the configuration of the system along with the solver which is to be used for solving.
+
+For example, consider the following configuration :
+
+```julia
+hamiltonian = sigmax                           # the Hamiltonian of the system
+initial_state = statevec(1, FiniteBasis(2))    # the initial state of the system
+tlist = 0.:0.1:2*pi                            # the time steps at each of which we have the evolved state
+```
+
+Now we have the basic ingredients to construct the `QuPropagator`, assuming we have a solver in mind.
+We construct the following :
+
+```julia
+qprop_ode45 = QuPropagator(hamiltonian, initial_state, tlist, QuODE45())
+```
+
+The general format of `QuPropagator` construct is as follows :
+`QuPropagator(Hamiltonian_of_the_system, Collapse_operators, Initial_state, Time_step_array, Solver)`
+
+Alternatively we can also pass :
+`QuPropagator(Type_of_Equation, Initial_state, Time_step_array, Solver)`
+
+Implying, we could do the following which is equivalent to the above construction of `qprop_ode45`
+
+```julia
+schrodinger_eq = QuSchrodingerEq(hamiltonian)
+qprop_ode45 = QuPropagator(schrodinger_eq, initial_state, tlist, QuODE45())
+```
+
+A common convention for the `equations` and `solvers` types and methods is they start with a prefix `Qu`.
+For example :
+```
+`Schrodinger Equation` => `QuSchrodingerEq`
+`ode45` => `QuODE45`
+`expmv` => `QuExpmV` or `QuExpokit` (based on the packages)
+```
+
+Now we are all set to calculate the states at various time steps which can be achieved in the
+following ways :
+
+**Method - 1**
+To get the evolved state at every time instant, we can do the following :
+```julia
+for (t, psi) in qprop_ode45
+    println("t : $t, psi : $psi")
+end
+```
+
+**Method - 2**
+To get the evolved state at the next time instant (here we get the state after the fist time-step), we can do the following :
+```julia
+next_state = next(qprop_ode45, start(qprop_ode45))
```

Comment:
Nope we do not export `propagate`. So do we go with this or export `propagate` and include it here ?